examples/07_join_aggregation.py

```diff
@@ -0,0 +1,438 @@
+
+ratings = fetch_movielens(dataset_id="ratings")
+ratings = ratings.X
+ratings
+
+
+# In[2]:
+
+
+import pandas as pd
+
+ratings["timestamp"] = pd.to_datetime(ratings["timestamp"], unit="s")
+ratings["year"] = ratings["timestamp"].dt.year
+ratings["day_name"] = ratings["timestamp"].dt.day_name()
+
+
+# ## Quick Data Exploration
+
+# In[3]:
+
+
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+sns.set_style("darkgrid")
+
+
+def make_barplot(x, y, title):
+    norm = plt.Normalize(y.min(), y.max())
+    cmap = plt.get_cmap("magma")
+
+    ax = sns.barplot(
+        x=x,
+        y=y,
+        palette=cmap(norm(y)),
+    )
+    plt.xticks(rotation=30)
+    plt.xlabel(None)
+    plt.ylabel(None)
+    plt.tight_layout()
+    plt.title(title)
+
+
+# In[4]:
+
+
+year_volume = ratings["year"].value_counts().sort_index()
+
+make_barplot(
+    x=year_volume.index,
+    y=year_volume.values,
+    title="Yearly volume of ratings",
+)
+
+
+# In[5]:
+
+
+ratings["day_index"] = ratings["timestamp"].dt.weekday
+
+day_name_index = (
+    ratings[["day_index", "day_name"]].drop_duplicates().set_index("day_index")
+)
+day_name_index.sort_index()
+
+
+# In[6]:
+
+
+daily_volume = (
+    ratings["day_index"].value_counts().sort_index().to_frame().join(day_name_index)
+)
+
+make_barplot(
+    x=daily_volume["day_name"], y=daily_volume["count"], title="Daily volume of ratings"
+)
+
+
+# In[7]:
+
+
+user_count_ratings = (
+    ratings.groupby("userId").agg(n_ratings=("rating", "count")).reset_index()
+)
+
+print(
+    "min:",
+    user_count_ratings["n_ratings"].min(),
+    "max:",
+    user_count_ratings["n_ratings"].max(),
+)
+
+ax = sns.histplot(user_count_ratings["n_ratings"])
+ax.set(
+    xlabel="Number of ratings given by users",
+    ylabel="Total number of users",
+    title="Number of ratings given by user",
+)
+
+
+# In[8]:
+
+
+rating_count = ratings["rating"].value_counts().sort_index()
+
+make_barplot(
+    x=rating_count.index,
+    y=rating_count.values,
+    title="Distribution of ratings given to movies",
+)
+
+
+# In[9]:
+
+
+## Split train test by date
+
+min_date = ratings["timestamp"].min()
+max_date = ratings["timestamp"].max()
+duration = max_date - min_date
+train_threshold = max_date - duration / 3
+
+train = ratings.loc[ratings["timestamp"] < train_threshold]
+test = ratings.loc[ratings["timestamp"] >= train_threshold]
+train.shape, test.shape
+
+
+# ## Feature Engineering
+
+# ***Ohe Hot Encoding day names***
+
+# In[10]:
+
+
+from sklearn.pipeline import make_pipeline, FunctionTransformer
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import OneHotEncoder
+
+col_trans = make_column_transformer(
+    ("passthrough", ["userId", "movieId", "rating"]),
+    (OneHotEncoder(sparse_output=False), ["day_name"]),
+    remainder="drop",
+    verbose_feature_names_out=False,
+)
+
+train = col_trans.fit_transform(train)
+train = pd.DataFrame(train, columns=col_trans.get_feature_names_out())
+train.head()
```

Comment:
In recent scikit-learn, you can probably replace the 3 lines above by something like (not tested):
```suggestion
train = col_trans.set_output(transform="pandas").fit_transform(train)
train.head()
```